docker/Dockerfile

```diff
@@ -8,10 +8,19 @@ ENV PATH $CASSANDRA_DIR/bin:$PATH
 
 # Untar our distribution
 ADD palantir-cassandra-*tgz /
+RUN mkdir $CASSANDRA_DIR/logs; \
+  mkdir -p $CASSANDRA_DIR/data/saved_caches; \
+  mkdir -p /var/lib/cassandra/persistent_settings; \
+  mkdir -p /var/lib/cassandra/data; \
+  mkdir -p /var/lib/cassandra/commitlog; \
+  mkdir -p /var/data/tmp; \
+  chown -R cassandra:cassandra /var/lib/cassandra /var/data/tmp $CASSANDRA_DIR
 
-# We give the cassandra user permissions to create things in the Cassandra dir
-RUN ["sh", "-c", "chmod +rwx ${CASSANDRA_DIR}"]
-RUN ["sh", "-c", "chown cassandra:cassandra ${CASSANDRA_DIR}"]
+
+# Install java 17
+RUN apt-get update && apt-get -y install openjdk-17-jdk
+ENV JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64/
+ENV PATH=$JAVA_HOME:$PATH
```

Comment:
Should we use corretto here? https://docs.aws.amazon.com/corretto/latest/corretto-17-ug/generic-linux-install.html

Reply:
Good question! Originally I tried to use Corretto, but it's not possible as they use alpine linux , which uses `musl` rather than `glibc` which is needed for some debugging purposes.